integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateUtil.scala

```diff
@@ -126,19 +126,33 @@ object PreAggregateUtil {
           attr.aggregateFunction,
           parentTableName,
           parentDatabaseName,
-          parentTableId)
+          parentTableId,
+          "column_" + counter)
+        counter = counter + 1
       case attr: AttributeReference =>
+        val columnRelation = getColumnRelation(attr.name,
+          parentTableId,
+          parentTableName,
+          parentDatabaseName,
+          carbonTable)
+        val arrayBuffer = new ArrayBuffer[ColumnTableRelation]()
+        arrayBuffer += columnRelation
         fieldToDataMapFieldMap += getField(attr.name,
           attr.dataType,
-          parentColumnId = carbonTable.getColumnByName(parentTableName, attr.name).getColumnId,
           parentTableName = parentTableName,
-          parentDatabaseName = parentDatabaseName, parentTableId = parentTableId)
+          columnTableRelationList = arrayBuffer.toList)
       case Alias(attr: AttributeReference, _) =>
+        val columnRelation = getColumnRelation(attr.name,
```

Comment:
move `attr.name` to next line, do the same for all place in this function

Reply:
I think `getColumnRelation` is no need, you can create a new ColumnTableRelation directly here, parameter  is almost the same